doc/CodeDocumentation.conf.in

```diff
@@ -1125,7 +1125,13 @@ IMAGE_PATH             =
 # need to set EXTENSION_MAPPING for the extension otherwise the files are not
 # properly processed by doxygen.
 
-INPUT_FILTER           =
+# Allow the use of standard LaTeX formulas in MFEM comments by replacing
+# $$...$$ with \f[...\f] and $...$ with \f$...\f$ before running Doxygen.
+# The regular expression ((?:.|\n)+?) is a lazy match for one or more
+# characters, include newline. If present, Doxygen-style LaTeX commands
+# such as \f$, \f[, etc., are left unchanged.
+
+INPUT_FILTER           = perl -0777 -pe 's/\$\$((?:.|\n)+?)\$\$/\\f[\1\\f]/g; s/(?<!\\f)\$((?:.|\n)+?)\$/\\f\$\1\\f\$/g'
```

Comment:
When I use an older `doxygen` (1.8.14) on macOS, I'm getting errors like these:
```console
...
Reading /Users/dobrev1/code/mfem/mfem.hpp...
sh: perl-0777-pes/\$\$((?:.|\n)+?)\$\$/\\f[\1\\f]/g;s/(?<!\\f)\$((?:.|\n)+?)\$/\\f\$\1\\f\$/g: No such file or directory
Parsing file /Users/dobrev1/code/mfem/mfem.hpp...
...
```
Putting double-quotes (`"`) around the command seems to fix this issue:
```
INPUT_FILTER           = "perl -0777 -pe 's/\$\$((?:.|\n)+?)\$\$/\\f[\1\\f]/g; s/(?<!\\f)\$((?:.|\n)+?)\$/\\f\$\1\\f\$/g'"
```
I'm not sure, but I assume newer `doxygen` doesn't have that issue on macOS.

Reply:
Also, with the older `doxygen` version, the math formulas are not displayed properly. If I revert the changes to the MATHJAX options, the math is fine.

Reply:
We can add the quotes, but overall I think it is OK to switch to the new Doxygen version -- the documentation is mostly build by us, and that requirement is [documented](https://github.com/mfem/mfem/blob/master/doc/README#L17)

Reply:
See #4149.